optimum/onnxruntime/modeling_decoder.py

```diff
@@ -394,15 +406,53 @@ def _from_pretrained(
 
             last_decoder_with_past_name = paths.get("last_decoder_with_past_model_name", None)
             if last_decoder_with_past_name is not None:
-                last_decoder_with_past_name = new_model_save_dir / last_decoder_with_past_name
-
-            model = cls.load_model(
-                decoder_path=new_model_save_dir / paths["last_decoder_model_name"],
-                decoder_with_past_path=last_decoder_with_past_name,
-                provider=provider,
-                session_options=session_options,
-                provider_options=provider_options,
+                decoder_with_past_path = new_model_save_dir / last_decoder_with_past_name
+
+            decoder_path = new_model_save_dir / paths["last_decoder_model_name"]
+
+        is_merged = has_onnx_input(decoder_path, "use_cache")
+        if is_merged is True:
+            decoder_with_past_path = None
+            if use_merged is False:
+                logger.info(
+                    "`use_merged` is automatically set to `True` as the decoder takes `use_cache` as an input."
+                )
+                use_merged = True
+        elif use_merged is True and decoder_with_past_path is not None:
+            try:
+                logger.info("Merging decoders...")
+                decoder_merged_path = decoder_path.parent / ONNX_DECODER_MERGED_NAME
+                merge_decoders(
+                    decoder=decoder_path,
+                    decoder_with_past=decoder_with_past_path,
+                    save_path=decoder_merged_path,
+                )
+                decoder_path = decoder_merged_path
+                decoder_with_past_path = None
+            except Exception as e:
+                logger.error(
+                    f"{e}.\nUnable to merge decoders. Will load two decoders for inference which will double the memory usage."
```

Comment:
This error message is used as several place, maybe make this a variable somewhere. This will make things easier if we need to update it.